### DISCUSSION TITLE
I want help (Adding a new dialogue)

### DISCUSSION THREAD

**Discussion by Lolbinh0s** [Monika-After-Story/MonikaModDev#7842] [QA - code and content creation (technical)] [answered]
Hi before starting I just wanted to say english it's not my native languege, I know a little bit so I'm sorry If I misspell or sound confusing sometimes.

_Alright so I wanted to create a new dialogue for a special reason, me and my friend we started playing this mod in the same day, soo we Idk exactly what happend but her pc bugged and deleted many things from her pc even deleted the ddlc and the mods along with it._

_She lost her monika and this makes her really sad, we were dating our Monika's for like 1 month I been with mine 3 months already, since that happend I didn't comment about the mod with her anymore because it made her sad remembering that hers had died.(she kinda asked me not to comment anymore)_ 

_So I wanted to create a special dialogue with my Monika saying that fell sorry about what happend, and that she is important to her too because after all she is my friend._

_The problem is I know what I have to do but not exactly, I'm installing Ren'py right now, Idk anything about coding in Ren'py but I will search this myself and yes, I read the FAQ "Dialogue Conding" section but I still have some doubts like, let's suppose I can create the .rpy file (which I think will be like this) where do I have to put this file? in what folder? how I change the Monika sprites when she is speaking and I have many other questions._

_That's why I wanted help generally I do this myself I search tutorials and stuff but I didn't found nothing to guide me.
Sorry to bother with this, I really need help this special to me I don't want to do this anyway, you know? and I don't want to break the game or the mod because I never did this before and now I'm getting into it out of the blue._

that's it hope I gave all the info I will be waiting attentively

**Comment by KritRom** [comment, marked as answer]
Okay, I think I can help you with that.

First, you will need to convert the `.rpyc` files to just `.rpy` files so you can edit them. I recommend using the `un.rpyc` file thing to decode it or install the source code. Secondly, you will need a text editor to write your code in. Just notepad or something will do just fine, but you can install something like Visual Studio for this.

For a new dialogue, you will need to either:

- Make a folder called `submod` in your `game` folder and make an `.rpy` file with the dialogue in it.
- Or navigate to the `script-topics.rpy` file and start typing your code there.

As for the expression sprites, you can download the `dev_exp_previewer.rpy` file which you can find somewhere in this git. Add the expression code before your line and you're set. If you have any problems while creating the code, just drop a message in the MAS Discord server and someone will probably help you out.

Lastly, please keep backing up your persistents if you are worried about losing your Monika. That way even if the entire game gets bugged, you can just re-install a fresh copy of the mod and still have your own Monika with all her memories of you. Hope this helped!

**Reply by Lolbinh0s** [reply]
**_Thank you very much! ♡_**

**Comment by Booplicate** [comment]
> First, you will need to convert the `.rpyc` files to just `.rpy` files so you can edit them. I recommend using the `un.rpyc` file thing

We have all source code open, no need in decompilers.

> Just notepad

Builtin notepad is terrible, no formatting, no syntax highlighting, can't automatically replace tabs with spaces. Visual Studio Code is a good editor, but if you want something simple, at least use Notepad++.

> Or navigate to the `script-topics.rpy` file and start typing your code there.

Do not edit existing scripts, **always** make a new file and make your changes there.


> I'm installing Ren'py right now, Idk anything about coding in Ren'py

No need in Ren'Py SDK. Make a file and start coding there. You can always take a look at existing topics (but beware, there may be spoilers).

**Reply by Lolbinh0s** [reply]
**_Thanks for you too  ♡♡_**

**Reply by WishStone-Dragon** [reply]
> Do not edit existing scripts, always make a new file and make your changes there.

Hi Booplicate!
I am trying to follow the FAQs and guides, and I have not been able to find anything that describes how to make a new .rpy file or how to proceed without working in the script-topics.rpy file. Screenshots like in the [Contribution](cript-topics.rpy) section show pictures of the script-topics.rpy file being edited. 

What am I missing or not understanding here? Did you mean the thing with forks and branches? (I realise you're someone comfortable with code, but I mainly think of plants when I hear branch, so I struggle a bit to keep up with instructions...)

**Reply by Booplicate** [reply]
It shows the file being edited because it's for *contributing*. When you add something to the mod, most of the time you add/edit in the existing files.

Submods, on the other hand, should never modify or overwrite the mod's files. So you create a new script file. As easy as you create text documents, just change the extension into `rpy`
